### PR TITLE
Fix issues with WebSocket onMessage and onClose callbacks on the Taobao Mini Game platform.

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #

### Changelog
https://github.com/cocos/cocos-engine/issues/17288
*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js [hunk 1]
```diff
@@ -55,10 +55,26 @@ export default class WebSocket {
     my.onSocketOpen(this._onOpen)
 
     this._onMessage = (res) => {
-      if (res && res.data && res.isBuffer) {
-        res.data = _utils.base64ToArrayBuffer(res.data)
+      function deepCopy(obj) {
+        if (typeof obj !== 'object' || obj === null) {
```
Comment (dumganhar) [inline platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js]: Use `obj == null` which will also support `undefined`.
Should it also need to check `undefined` value?

File: platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js [hunk 2]
```diff
@@ -55,10 +55,26 @@ export default class WebSocket {
...
+          if (obj.hasOwnProperty(key)) {
+            copiedObj[key] = deepCopy(obj[key]);
+          }
+        }
+      
+        return copiedObj;
+      }
+      let tmp = deepCopy(res);
+      if (tmp && tmp.data && tmp.isBuffer) {
+        tmp.data = _utils.base64ToArrayBuffer(tmp.data)
```
Comment (dumganhar) [inline platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js:75]: So taobao platform only supports `string` response content?

File: platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js [hunk 3]
```diff
@@ -69,10 +85,10 @@ export default class WebSocket {
     my.onSocketError(this._onError)
 
     this._onClose = (res) => {
-      this.readyState = WebSocket.CLOSED
-      this._triggerEvent('close')
-      this._removeAllSocketListenr()
+      this.readyState = WebSocket.CLOSED;
       this._decreaseCount();
+      this._removeAllSocketListener();
+      this._triggerEvent('close');
```
Comment (dumganhar) [inline platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js:91]: Does the order of removing listeners and firing close event matter?

File: platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js (hunk 1 above)
Comment (bofeng-song) [reply platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js]: Yes, need to check undefined value.
image

File: platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js (hunk 3 above)
Comment (bofeng-song) [reply platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js:91]: Due to the singleton nature of the class, failing to call decrease before callback will result in the creation of a new WebSocket instance within the callback, which will fail.

File: platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js (hunk 2 above)
Comment (bofeng-song) [reply platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/WebSocket.js:75]: According to Taobao's official requirements, ArrayBuffers must be encoded in Base64 format.

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**